src/devices/DCMotorController/DCMotorController.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Pwm;
+using System.Device.Pwm.Drivers;
+
+namespace Iot.Device
+{
+    public class DCMotorController : IDisposable
+    {
+        private const double PwmFrequency = 50; // Hz
+        private PwmController _pwm;
+        private GpioController _controller;
+        private int _d0;
+        private int? _d1;
+        private double _speed;
+
+        /// <summary>
+        /// Initialize a DCMotorController class
+        /// </summary>
+        /// <param name="pin0">The GPIO pin number used for controlling the speed of the motor.</param>
+        /// <param name="pin1">The optional GPIO pin number used for controlling the direction of the motor.</param>
```

Comment:
Should we add some instruction as well mentioning what would a PinValue.High or a PinValue.Low would mean in terms of direction? Also, should we provide what is the default when null is used?

Reply:
I'll actually change this completely. It'll be more like:
`int? d0, int? d1, int? enable` and this basically will be basically asking the user how did they connect the motor to the H-bridge (naming matches the pins on H-bridge) and depending on how they connected we will internally change behavior to work with that